src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -656,6 +656,11 @@ private BoundExpression BindExpressionInternal(ExpressionSyntax node, Diagnostic
             }
         }
 
+        private static BoundExpression BindDefaultLiteral(DefaultLiteralSyntax node)
+        {
+            return new BoundDefaultLiteral(node, ConstantValue.DefaultLiteral, type: null);
+        }
```

Comment:
What about scenarios when it is of type user-defined struct? Is it supposed to have constant value then?